minject/inject.py

```diff
@@ -91,6 +120,23 @@ def _is_type(key: "RegistryKey[T]") -> TypeGuard[Type[T]]:
     return isinstance(key, type)
 
 
+def _is_key_async(key: "RegistryKey[T]") -> bool:
+    """
+    Check whether a registry key is an async context manager
+    marked for initialization within the registry (@async_context).
+    """
+    if isinstance(key, str):
+        return False
```

Comment:
This seems reasonable, however I'd like to defer making this edit for the sake of time constraints surrounding the project.

I've made an issue documenting this suggestion, I think it could make sense to group this with other Refactor suggestions and go through them together when we have time (most likely next grease-week/hackathon).